Add planemo-mode

mtekman opened the pull request:
### Brief summary of what the package does

This package provides a minor mode for developing tool wrappers for the [GalaxyProject](https://galaxyproject.eu/), which makes use of a [specific XML-based schema](https://docs.galaxyproject.org/en/master/dev/schema.html) in conjunction with the [Cheetah template engine](https://cheetahtemplate.org/).

The mode brings Bash and Cheetah fontification (inspired by the [Cheetah Mode](https://www.emacswiki.org/emacs/CheetahMode) in the Emacs Wiki) and indentation functions. 

### Direct link to the package repository

https://gitlab.com/mtekman/planemo-mode.el

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Thanks for this, code looks good.  I think the summary line should call this a major mode (not a minor mode).  Some quick feedback:

- I notice your customizable variables are all set `--private` but customizable variables are by definition part of a package's public interface (accessible to even novice users via <kbd>M-x customize-group</kbd>).
- For the sake of performance you should know `line-number-at-pos` steps linearly through the buffer, starting from the top.  Depending on how big planemo files can get, how often this function will get called, and how much you care :) you may want to think about alternative implementations of `planemo--numlines` that don't call this function
- planemo-mode.el#L206: You can use `not` or `null`

mtekman commented:
@riscy thanks for the review! I've made the changes to the customisation functions and fixed that condition you mentioned. 

Thanks for telling me about the `line-number-at-pos` function being sub-optimal, for some reason I just assumed that it was reading off a global line number variable. I've fixed it now to use the `(count-matches "\n")` function which hopefully should be a bit better.

Cheers!

riscy commented:
No linter catches this but the docstring e.g. for `planemo-root-align-region` should probably be wordwrapped (<kbd>M-q</kbd>).  Surprisingly M-x apropos will wordwrap (imo it shouldn't), but the customization interface and help-variable interfaces won't , leading to bad display.  Non-blocking, though.  :)
